### DISCUSSION TITLE
im871a and Lora support?

### DISCUSSION THREAD

**Discussion by PovilasID** [wmbusmeters/wmbusmeters#1108] [Ideas]
Hello,

I my utility company has installed some meters that transmit Lora signals on 868 Mhz band  and meter manufacturer said that packets are undecrypted (I know... I know... any way..) 

Theoretically it should be possible to capture Lora signal but will it show up in the list of received telegrams? 
Is there then some im871a compatible tool that could work? And can two software tools (wmbusmeter + lora packet capturer) work at the same time addressing the same device? (I think im871a  has some back and forth communication...)

**Comment by weetmuts** [comment]
Interesting! If its unencrypted, then I assume it is not integrated with some LPWAN (lora to gateway to internet server etc).
We could actually listen to this, if this is the case. I have added some rudimentary support already in wmbusmeters for talking to an lora iu880b dongle. 

I postponed the lora work when I realized that a lot of lora meters send encrypted signals for the lpwan gateway/server decryption which is definitely impossible to get hold of. But if you can get confirmation from the meter manufacturer that the content is available to anyone with a lora receiver, then we can have a go at it.

Which meter is it?

**Reply by PovilasID** [reply]
They do collect the data remotely but distributor said they are undecrypted... I have not used Lora before, so I have no idea what to even ask about it. I am running my im871a  without meters defined so it lists all heard telegrams... I am not sure if Lora signal would show up in that list and if I am even close enough to pick it up. (Lora theoretical max is ~5km and I am very close compared to it but antena in the USB may not be suitable) 

In about a week I should get Nooelec NESDR Mini 2+ that should be able to pick up all lose signals. I was planning on using to do some more investigations.

Meter: QALCOSONIC E3

**Reply by PovilasID** [reply]
UPDATE: TL;DR No progress. 

I got my SDR receiver and set it up to pick up signals on 868.1 Mhz which is supposed to be Lora band but I am getting buried in signals. I am not sure at what I am looking here maybe I misconfigured something... Keep in mind this is the first time touching SDR. 
![image](https://github.com/wmbusmeters/wmbusmeters/assets/396243/8aa49527-7704-499b-9786-f4ad6ed320bd)
However, there is good news too. I managed to fish out both Lora packet specification 
[Lora packet specification](https://github.com/wmbusmeters/wmbusmeters/files/13516794/Axioma_Lora_Payload_E3_2V00_Extended.converted.pdf) and a [nodered script decoding it](https://gist.github.com/Alkarex/4b5d1fef2ff84d483e2793ed009ef607), so if I actually manage to capture some packets I should be able to read it.

**Reply by PovilasID** [reply]
UPPDATE 2: TL;DR Captured signals not sure how to get hex out of them.

I was able to tune SDRangel to capture some what looked liked actual lora signals however text TXT portion was always a mess, so I presume I was way off in my configuration. Hence, having no clue how to fix it went to see if I am even getting lora signals at all.  Here is a sample of one of the better signals...
```
>00:34:05 b6 S:-22.4 SN:0.8 HF:err HC:err FEC:fix CRC:err
000|da cb 17 5c|3e 23 4e d9|58 af 88 97|0b 39 32 dd
016|f8 e6 5d 0b|be ae 6f a2|f5 5d f5 e4|34 92 a7 d3
032|c6 33 77 cf|db b5 13 72|5d 52 48 0d|e5 11 2e 4a
048|cf a7 b9 3f|67 5f e7 d1|ee e9 94 6d|4a 09 1a 16
064|8e e9 d8 04|9b e7 16 c9|73 44 2d 6c|96 ce 05 e3
080|a0 c1 4e a9|c9 5d 97 37|d4 91 57 93|18 d4 9d 3d
096|e4 34 4a f0|15 95 a0 23|12 a4 79 04|78 3c c0 6e
112|85 3a 8f 30|a6 85 64 9d|b7 45 7b cc|e0 6e 93 16
128|8a 40 d9 bc|a2 83 39 b4|86 64 a5 92|44 94 3c e2
144|6f 99 2e 39|31 be ac 98|6b b4 55 63|7c e2 86 fc
160|c5 a9 c3 6f|6b 04 28 11|8b 3a e5 ef|58 9c 2c bd
176|9c 0e e0 1e|ca d9 c2 94|09 33 ad eb|08 70 7a 4f
192|3d 25 8c 62|ab 9c 18 8f|33 95 2b 52|f3 e6 8e 05
208|be 3b 0a 1e|af db 39 49|7f 5a e9 95|e5 8d 
TXT|��\>#N�X���92���]��o��]��4����3w�۵r]RH
�.Jϧ�?g_�����mJ	������sD-l�����N��]�7ԑW�ԝ=�4J���#�yx<�n�:�0��d��E{��n��@ټ��9��d��D�<�o�.91���k�Uc|���ũ�ok(�:��X�,�����	3��pzO=%�b���3�+R����;
��9I
```
So I switched to using rtl_433 tool for capturing signals and looks like I have captured some clean Lora signals  (not the same as above):
![image](https://github.com/wmbusmeters/wmbusmeters/assets/396243/41c6c1d7-c2fb-4769-b97f-a5a3550fb1bf)
[plausible.zip](https://github.com/wmbusmeters/wmbusmeters/files/13536185/plausible.zip) ( Files can be looked at using https://triq.org/pdv )
I also found and started running a tool called gr-lora to try turn those signals into something processable but since I do not have a transmitter I am just sitting fishing for those signals. Considering I have no clue if my setup is correct... this may be a very slow debugging process.

**Reply by PovilasID** [reply]
TL;DR Possible but not pratical.
I ran into an wall... well couple. 
a) The meter that I was looking to decode Lora signal for... does not actually have a lora module in it... at all. I was getting LoraWan signals.
b) Most other meters are using LoraWAN protocol. LoraWAN signals are encrypted and there is join procedure to the network that kicks out other networks aka your utils provider will not see the data. Here is a good write up https://penthertz.com/blog/testing-LoRa-with-SDR-and-handy-tools.html
It is possible to capture and process data using RTL-SDR however it is not practical if it is using LoraWAN and is on the utility provider's network. Does not mean impossible it just much more work to turn research tools into production and also you need transmission capabilities. However, if your lora signal is not encrypted I would recommend looking into https://www.lilygo.cc/products/lora3 these are all in one solutions and are very cheap. I have not tested them but they look grate. Would be awesome to see wmbusmeter running on them.
